packages/core/lib/commands/compile.js

```diff
@@ -6,19 +6,19 @@ const command = {
   builder: {
```

Comment:
So we're currently not sure what this field does? I think it may help yargs to parse options? I guess we should see what happens when we remove them :)

Reply:
Anyways that is unrelated to this PR.

Reply:
Yeah, of course unrelated, but @eggplantzzz I spent a fair bit of time trying to find a place where it's actually used but I couldn't find anything. Maybe you can take a second look?

Reply:
So it does provide defaults to the config object. So if you print the config after it is created you get defaults for all the options  listed in `builder` as it sort of gets merged in somehow. So check out the end of the config when I print it after executing `truffle compile --all`:
```
 timeoutBlocks: [Getter/Setter],
  _: [ [length]: 0 ],
  all: true,
  help: false,
  compiler: null,
  list: undefined }
```

If you remove `builder` then you don't get the defaults for `help`, `compiler`, and `list` like so:
```
 timeoutBlocks: [Getter/Setter],
  _: [ [length]: 0 ],
  all: true }
```

Maybe we should remove this stuff unless it ends up causing some problem in CI. Although maybe we should keep it because it allows us to have defaults per command.